### PR TITLE
chore: add assertions for router transform and batch outputs

### PR DESCRIPTION
**Pull request by chandumlg**
## What are the changes introduced in this PR?

add assertions for router transform and batch outputs

## What is the related Linear task?

Resolves [INT-375](https://linear.app/rudderstack/issue/INT-375/fix-the-batching-order-in-the-applicable-routertransform-destinations) [INT-1298](https://linear.app/rudderstack/issue/INT-1298/enhance-component-test-suite-to-do-server-contract-validations-for)

## Please explain the objectives of your changes below

Adding asserts to router transform and batching outputs.
https://linear.app/rudderstack/issue/INT-375/fix-the-batching-order-in-the-applicable-routertransform-destinations

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: test/__tests__/lambda.test.js [hunk 1]
```diff
@@ -40,6 +41,7 @@ describe(`${name} Tests`, () => {
   describe("Router Tests", () => {
     it("Payload", async () => {
       const routerOutput = await transformer.processRouterDest(inputRouterData);
+      assertRouterOutput(routerOutput, inputRouterData);
```
Comment (koladilip) [inline test/__tests__/lambda.test.js:44]: @ItsSudip why are these files still present? didn't we move them to component testing?

File: test/testHelper.js [hunk 2]
```diff
@@ -15,11 +15,81 @@ const setResponsesForMockAxiosAdapter = ({url, method, data, options}, {response
...
+        const jobId = meta.jobId;
+        const userId = meta.userId;
+        let arr = userIdJobIdMap[userId] || [];
+        arr.push(jobId);
+        userIdJobIdMap[userId] = arr;
+      });
+    }
+  });
+
+  console.log(userIdJobIdMap);
```
Comment (koladilip) [inline test/testHelper.js:71]: remove this

**Review by krishna2020 — DISMISSED** [review]

**Comment by koladilip** [comment]
@chandumlg are you using a different formatter by any chance?

**Comment by chandumlg** [comment]
> @chandumlg are you using a different formatter by any chance?

Im using prettier only.

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2982/test-report.html

File: test/__tests__/lambda.test.js (hunk 1 above)
Comment (chandumlg) [reply test/__tests__/lambda.test.js]: Rebased this pr onto develop after Sudip's PR is merged.

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2982/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2982/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2982/test-report.html

File: test/__tests__/data/pinterest_tag_router_batch_output.json
Comment (koladilip) [inline test/__tests__/data/pinterest_tag_router_batch_output.json:1]: are we using them?

**Review by koladilip — APPROVED** [review]

File: test/__tests__/data/pinterest_tag_router_batch_output.json
Comment (chandumlg) [reply test/__tests__/data/pinterest_tag_router_batch_output.json:1]: Yes. In pinterest component test.

**Review by sanpj2292 — APPROVED** [review]

File: test/__tests__/data/af_router_input.json
[the remark this answers is not in the document]
Comment (chandumlg) [reply test/__tests__/data/af_router_input.json:1]: Rebased after removing them.

**Pull request merged**